server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -5507,6 +5647,27 @@ public void updateShardIngestionState(IngestionStatus ingestionStatus) {
         }
     }
 
+    /**
+     * Drains all ReleasableRetryableRefreshListener instances to ensure ongoing refresh operations complete
+     * before shard shutdown. This is primarily used in test scenarios.
+     *
+     * @return List of Releasable objects that must be closed to release the drained permits
+     */
+    public List<Releasable> drainRefreshListeners() {
```

Comment:
Lets see if this can be called during the close method invocation in IndexShard. This will reduce the number of times this method needs to be called for tests.